### PR TITLE
Adding coverage for cap=1 in communication baseline test

### PR DESCRIPTION
**Pull request by benjaminsavage**

### CODE REVIEW

File: src/protocol/ipa/mod.rs [hunk 1]
```diff
@@ -963,79 +963,96 @@ pub mod tests {
...
+                        NUM_MULTI_BITS,
+                    )
+                    .await
+                    .unwrap()
+                })
                 .await
-                .unwrap()
-            })
-            .await;
+                .reconstruct();
```
Comment (benjaminsavage) [inline src/protocol/ipa/mod.rs:1008]: This is all just indented a bit more... I wish diff view was smarter...

File: src/protocol/ipa/mod.rs [hunk 2]
```diff
@@ -963,79 +963,96 @@ pub mod tests {
...
+            let snapshot = world.metrics_snapshot();
+            let records_sent = snapshot.get_counter(RECORDS_SENT);
+            let semi_honest_baseline = if per_user_cap == 1 {
+                RECORDS_SENT_SEMI_HONEST_BASELINE_CAP_1
+            } else {
+                RECORDS_SENT_SEMI_HONEST_BASELINE_CAP_3
+            };
+            assert!(records_sent <= semi_honest_baseline,
+                "Baseline for semi-honest IPA (cap = {per_user_cap}) has DEGRADED! Expected {semi_honest_baseline}, got {records_sent}.");
+
```
Comment (benjaminsavage) [inline src/protocol/ipa/mod.rs:1019]: Added this nice error message for the semi-honest case. We only had it for the malicious case for some reason.

File: src/protocol/ipa/mod.rs [hunk 3]
```diff
@@ -963,79 +963,96 @@ pub mod tests {
...
-        assert!(records_sent <= RECORDS_SENT_SEMI_HONEST_BASELINE);
-        if records_sent < RECORDS_SENT_SEMI_HONEST_BASELINE {
-            tracing::warn!("Baseline for semi-honest IPA has improved! Expected {RECORDS_SENT_SEMI_HONEST_BASELINE}, got {records_sent}.\
-                            Strongly consider adjusting the baseline, so the gains won't be accidentally offset by a regression.");
-        }
+        for per_user_cap in [1, 3] {
+            let world = TestWorld::new_with(*TestWorldConfig::default().enable_metrics()).await;
 
-        let world = TestWorld::new_with(*TestWorldConfig::default().enable_metrics()).await;
+            let records: Vec<GenericReportTestInput<Fp32BitPrime, MatchKey, BreakdownKey>> = ipa_test_input!(
```
Comment (taikiy) [inline src/protocol/ipa/mod.rs:984]: nit: This could move out from the for loop

**Review by taikiy — APPROVED** [review]
Just one nit comment. LGTM

File: src/protocol/ipa/mod.rs (hunk 2 above)
Comment (richajaindce) [reply src/protocol/ipa/mod.rs:1019]: That's my fault :/ I just added for malicious :(

**Pull request merged**